Update aws-eks.mdx

JZambranoslty opened the pull request:
Revamped deployment page

JZambranoslty committed:
Update aws-eks.mdx

Revamped deployment page

Signed-off-by: Julio Zambrano <101874589+JZambranoslty@users.noreply.github.com>

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
-- An AWS account.
+- An AWS account where you have admin rights (no need to create IAM Users, you will use your Admin credentials for the account).
+- A `fqdn` OR A valid hosted zone where you can create a new record.
+  - You can validate if your hosted zone is exposed by running `dig <your hostedzonedomain>` in a terminal with Internet access. This should return `NOERROR` and its authority.
```
> This should return `NOERROR` and its authority.

This might be a bit confusing, because the default dig command's output is rather verbose.
We can suggest something like this:

`dig +stats +short <YOURHOSTEDZONE> NS`

I returns something or nothing depending on the status:

```bash
❯ dig +stats +short mostly.ai NS
ns-1939.awsdns-50.co.uk.
ns-1041.awsdns-02.org.
ns-437.awsdns-54.com.
ns-1014.awsdns-62.net.
❯ dig +stats +short mostlyASD.ai NS
```

JZambranoslty replied:
Corrected

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
 
-<Image src="/docimages/install/aws-eks/05-user-06-result.webp" alt="AWS IAM - new user added" width={800} height={300} />
+Go to your AWS Access Portal.
+From the Account where you have **AdministratorAccess** and you will use to create the cluster and deploy the platform, click on **Access keys**.
```
This is how our SSO is setup :) On the reader's end they won't have it or if they do - the setup is highly likely to be different.
Sending the reader to AWS's pages might be beneficial, e.g. here: <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>

JZambranoslty replied:
Corrected guiding to aws and clarifying that this is a SSO example

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+From the **Option 1: Set AWS environment variables**, click the button to copy your key, secret and token. Open a terminal in your IDE where you already have the MOSTLY AI helm charts available and paste them there.
 
-### Task 5: Create an access key for the user
+`#export <your aws key>`
```
If we're being this verbose, I'm pretty sure you meant

```bash
export AWS_ACCESS_KEY_ID=<YOURAWSKEY>
...
```

JZambranoslty replied:
Corrected

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
-   <Image src="/docimages/install/aws-eks/06-user-access-key-04-download-access-key.webp" alt="AWS IAM - Download access key" width={800} height={300} />
 
-**Result**
+Make sure to **also save your credentials in the aws cli** via `aws configure` (hit <Enter> after updating the values in every line to save the parameters):
```
Is there any reason for doing this even if we exported those previously? I'd suggest either one or the either, don't see a reason in doing both

JZambranoslty replied:
Yep, if you don't save your token, eksctl will complain about being unable to find it.

dmitrii-parshenkov replied:
In that case I'd just suggest setting the profile up instead directly with the aws cli configure maybe? Wdyt?

JZambranoslty replied:
You got a good point.
aws configure does not ask for the token that eksctl requires; and if I guide the user to edit their ~/.aws/credentials, the token is not even there. It all is consumed from the export lines.

I can just add the line export AWS_REGION="eu-central-1" and that works:
`julio@MOAI-MAC-10 helm-charts-5.0.3 % aws configure
AWS Access Key ID [****************56SG]: 
AWS Secret Access Key [****************K0gK]: 
Default region name [eu-central-1]: 
Default output format [json]: 
julio@MOAI-MAC-10 helm-charts-5.0.3 % export AWS_REGION="eu-central-1"
julio@MOAI-MAC-10 helm-charts-5.0.3 % cat ~/.aws/credentials          
[438217566446_AdministratorAccess]
aws_access_key_id=ASIAWMB...
aws_secret_access_key=rKUp96vxW...
aws_session_token=IQoJb3JpZ2luX2VjEBEaDGV1LWNlbnRyYWwtMSJIMEYCIQC7xtx+ZlEbbTsF9GA+4EgaNBlMN+QNh4sDjnzECOiyegIhAIM...
[default]
aws_access_key_id = ASIAWMB6NNDXN6JA56SG
aws_secret_access_key = rKUp96vxWBTMtjLjW4YNPbSD355qgkHcb6AjK0gK
julio@MOAI-MAC-10 helm-charts-5.0.3 % vi ~/.aws/credentials        
julio@MOAI-MAC-10 helm-charts-5.0.3 % cat ~/.aws/credentials
julio@MOAI-MAC-10 helm-charts-5.0.3 % eksctl create cluster -f mostlyai-cluster.yaml --dry-run
accessConfig:
  authenticationMode: API_AND_CONFIG_MAP
addons:
...
julio@MOAI-MAC-10 helm-charts-5.0.3 % aws sts get-caller-identity                             
{
    "UserId": "AROAWMB6NNDXKZNLVO6FQ:julio.zambrano@mostly.ai",
    "Account": "438217566446",
    "Arn": "arn:aws:sts::438217566446:assumed-role/AWSReservedSSO_AdministratorAccess_936009d95d56cff9/julio.zambrano@mostly.ai"
}
julio@MOAI-MAC-10 helm-charts-5.0.3 % `

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+    metadata:
+      name: mostly-ai
+      region: eu-central-1
+      version: "1.32"
```
If you drop the version, does it just use the latest? (Would be 1.33 at the time of writing). If yes - then use that instead, we always support latest Kubernetes

JZambranoslty replied:
Yes, dropping it goes for the latest. I just tried to have some safety net but it's redundant

JZambranoslty replied:
Edit: Dropping defaults to 1.32
<img width="2560" height="719" alt="Screenshot 2025-08-19 at 10 24 11 a m" src="https://github.com/user-attachments/assets/acce5566-5013-4401-bb0f-08780a395afe" />

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+      withOIDC: true
+    
+    addons:
+      # REQUIRED for podIdentityAssociations below
```
This being an example I don't think needs a comment of something being required for something. It's just an example and not a guideline per say, and if our reader questions the pod-identity-agent requirement they should be able to figure out why it's needed

JZambranoslty replied:
Dropped

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+            permissionPolicyARNs:
+              - arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy
+    
+    cloudWatch:
```
Is this something that's required? I'd drop all the cloudwatch configuration, it's not part of our requirements anyway

JZambranoslty replied:
Nope, just for visibility. Dropped

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+    managedNodeGroups:
+      - name: system-nodes
+        amiFamily: Bottlerocket
+        instanceTypes: [ m5.large ]
```
m5a instances are cheaper and we don't need heavy compute for system

JZambranoslty replied:
Corrected

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
```diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+        volumeType: gp3
+        volumeSize: 24
+    
+      - name: frontend-nodes
```
frontend-nodes is very confusing, and does not reflect their purpose. Honestly, I'd just leave two node types:

1. General nodes (product + system)
2. Compute nodes

2 m5a.xlarge instances should be more than plenty for General.

JZambranoslty replied:
Dropped in favour of a m5a.xlarge system-node

dmitrii-parshenkov commented on pages/install/deploy/aws-eks.mdx:
````diff
@@ -8,536 +8,294 @@ import { Callout } from 'nextra/components';
+   ```
 
-For details on how to configure each, see [_Ingress controllers_](../config/ingress-controllers).
+2. Create the default storage class to be used by the deployment:
````
I'm still not so sure on this being needed. May I ask you to deploy a testing cluster without gp3 SC and try running with the default SC -> reach out to me if anything fails? I'll be able to dig through.

_**You could**_ run into issues with the PostgreSQL PVC. These can be fixed with setting `mostlyPsql.deployment.enableEbsPVCBasedPermissions` to `true` in the values (assuming we're using mostly-combined)

JZambranoslty replied:
No issues with the pvs / pvcs, but mostly-app, keycloak and llm-proxy are unable to start for no apparent reason.
Everything is default to gp2 in the eksctl.yaml and values.yaml.
<img width="1656" height="537" alt="Screenshot 2025-08-19 at 11 01 22 a m" src="https://github.com/user-attachments/assets/e91c61bd-69e7-4022-8e8c-f35ee564f677" />


Attached logs
<img width="1654" height="533" alt="Screenshot 2025-08-19 at 10 55 20 a m" src="https://github.com/user-attachments/assets/144b7e9a-33d6-4ca6-9ba8-9fd6540568cc" />
<img width="1655" height="549" alt="Screenshot 2025-08-19 at 10 55 10 a m" src="https://github.com/user-attachments/assets/42ae39ec-a545-4d42-b522-8bd790236570" />
[llm-proxy.log](https://github.com/user-attachments/files/21852122/llm-proxy.log)
[keycloak.log](https://github.com/user-attachments/files/21852123/keycloak.log)
[mostly-app.log](https://github.com/user-attachments/files/21852124/mostly-app.log)
---
Nodes (1 m5a.xlarge for system and 1 c5.4xlarge for compute) don't seem to be under any pressure at all
<img width="1650" height="539" alt="Screenshot 2025-08-19 at 10 57 57 a m" src="https://github.com/user-attachments/assets/b75d50fd-d913-4424-a02b-71b538fed380" />

I could use your eyes here when you get a chance. The cluster is up in the test-saas account @dmitrii-parshenkov

JZambranoslty replied:
>2 m5a.xlarge instances should be more than plenty for General. 
The pods failed to be scheduled due to the lack of compute resources. Why not one m5a.2xlarge that has twice the capacity of one m5a.xlarge?

JZambranoslty replied:
And changing to only one m5a.2xlarge allows for all of the application pods to be scheduled; while the engine jobs respect their taints. 🥳 
<img width="2475" height="1012" alt="Screenshot 2025-08-19 at 12 29 31 p m" src="https://github.com/user-attachments/assets/1e9fac07-8553-4fca-9ecf-44b8fc11c674" />

JZambranoslty committed:
Update aws-eks.mdx

The one thing that's not working is that the mostly-app, keycloak and llm-proxy pods are unable to start and I can't find a clear reason yet.

JZambranoslty committed:
Update aws-eks.mdx

Removed gp3 references and bumped from 2 m5a.xlarge to one m5a.2xlarge for the system compute.
Necessary to explicitly map 1.33 as k8s version or it would default to 1.32.

JZambranoslty committed:
Update aws-eks.mdx

Removed redundant credentials saving

JZambranoslty committed:
Update aws-eks.mdx

Removed inexistent links

mostlyken committed:
lint

mostlyken committed:
syntax and style
